java/org/apache/catalina/TomcatPrincipal.java

```diff
@@ -47,4 +48,37 @@
      *                   exception to LoginContext
      */
     void logout() throws Exception;
+
+    /**
+     * Returns the value of the named attribute as an <code>Object</code>, or
+     * <code>null</code> if no attribute of the given name exists, or if
+     * <code>null</code> has been specified as the attribute's name.
+     * <p>
+     * Only the servlet container may set attributes to make available custom
+     * information about a Principal or the user it represents. For example, some of
```

Comment:
I think the For example is not ncessary, since we don't want to incline to any actual implementation.

Reply:
Will remove that.